Source/missiles.cpp

```diff
@@ -193,78 +199,31 @@ bool MonsterMHit(int pnum, int m, int mindam, int maxdam, int dist, missile_id t
 		return false;
 
 	int hit = GenerateRnd(100);
-	int hper = 0;
-	if (pnum != -1) {
-		const auto &player = Players[pnum];
-		if (MissilesData[t].mType == 0) {
-			hper = player.GetRangedPiercingToHit();
-			hper -= player.CalculateArmorPierce(monster.mArmorClass, false);
-			hper -= (dist * dist) / 2;
-		} else {
-			hper = player.GetMagicToHit() - (monster.mLevel * 2) - dist;
-		}
-	} else {
-		hper = GenerateRnd(75) - monster.mLevel * 2;
-	}
-
+	int hper = PlayerMissileCalculateCTHAgainstMonster(pnum, t, dist, monster);
 	hper = clamp(hper, 5, 95);
 
-	if (monster._mmode == MonsterMode::Petrified)
-		hit = 0;
-
 	if (monster.TryLiftGargoyle())
 		return true;
 
-	if (hit >= hper) {
+	if (hit >= hper && monster._mmode != MonsterMode::Petrified) {
 #ifdef _DEBUG
 		if (!DebugGodMode)
 #endif
 			return false;
 	}

```

Comment:
This block could be better isolated:

```cpp
if (monster.TryLiftGargoyle())
	return true;

if (monster._mmode != MonsterMode::Petrified) {
	int hit = GenerateRnd(100);
	int hper = PlayerMissileCalculateCTHAgainstMonster(pnum, t, dist, monster);
	hper = clamp(hper, 5, 95);

	if (hit >= hper) {
	#ifdef _DEBUG
		if (!DebugGodMode)
	#endif	
			return false;
	}
}
```
(I'm aware that this moves when RNG is fired, but that isn't an issue here)

Maybe we could even extract this block at some point:
```cpp
bool DidMissilesHit(int pnum, missile_id t, dist, Monster &monster)
{
#ifdef _DEBUG
	if (DebugGodMode)
		return true;
#endif	

	if (monster._mmode == MonsterMode::Petrified)
		return true;

	int toHit = CalculateMissileToHit(pnum, t, dist, monster);

	return GenerateRnd(100) < clamp(toHit, 5, 95);
}

if (monster.TryLiftGargoyle())
	return true;

if (!DidMissilesHit(pnum, t, dist, monster))
	return false;
```

Reply:
As a consequence of fixing bugs the exact sequence will change. This is also why two versions of the original are not able to join the same game. If someone wants things to be identical to 1.09 then they should use 1.0.x where no such changes had been made. The RNG can be manipulated by attacking the air, or by simply standing still in some cases, so the exact sequence isn't stable and shouldn't be though of as part of the game mechanics. For places where RNG needs to remain stable we write tests to asset this.

Reply:
Latest release for 1.0.x is 1.0.3, which has backports from 1.1.0:
https://github.com/diasurgical/devilutionX/releases/tag/1.0.3
The v1.0 branch also has backports from 1.2.0, but no release was created as download numbers are low and the double release confuses some users:
https://github.com/diasurgical/devilutionX/tree/v1.0

Clean up and refactoring of v1.0 is not intended. As things stands more then anything it's missing someone willing to backport bits from 1.3.0 and 1.4.0. I think Blighthorn from Discord also had some requests, some of which might be acceptable. So feel free to work on that if you like.